### ISSUE TITLE
No logging is done after an --exit

### ISSUE THREAD

**Issue by shazron** [phonegap/ios-sim#167]
Unlike ios-sim 3.x, we can't log after the process exits.

**Comment by shazron** [comment]
Alternatives: 
1. symlink to actual log 
2. some out-of-process logging to file

**Comment by haxpor** [comment]
Is this for logging during running of application as well? I cannot see logging message while running.

**Comment by shazron** [comment]
Yup logging for as long as the simulator is running, for the app. In 3.x even with --exit flag, you could log to file, but with 4.x, you can't.

**Comment by haxpor** [comment]
@shazron Alright! Thank you for info. I'm looking forward to this too.

**Comment by schlachtzeuger** [comment]
Hey all,

since I spent several days with this bug, I wanted to share my thoughts and my workaround/solution for this with you. 

Even if you omit the --exit flag and send ios-sim to background using &, the logging isn't done right but old log entries can suddenly appear in the log output. The culprit for this isn't clear to me, as fs.watch in the Tail module gets a "rename" event even if the system.log of the emulator wasn't renamed. When that event was raised, all contents of the simulator's system.log gets copied into the log output again. So maybe it's an issue within the fs module, but I don't know.

In our scenario, we run unit tests with Jasmine in the emulator and need to wait for a specific line to determine if the tests were successful or not. My workaround is this little script:

`
#!/bin/bash

SIM_DEVICE="iPhone 5s"
SIM_IOS="9.1"

# modify device name for ios-sim
MOD_DEVICE=`echo $SIM_DEVICE | sed 's/ /-/'`

### check if there exists a simulator for the given DEVICE_NAME and IOS_VERSION
SIM_UDID=""

while read -r line
do
    if [[ "$line" =~ "${SIM_DEVICE} (${SIM_IOS})" ]]; then
	SIM_UDID=`echo $line | sed 's/^.*\[\(.*\)\]$/\1/'`
    fi
done < <(instruments -s devices)

if [ "$SIM_UDID" = "" ]; then
    echo "no matching device found"
    exit 1
fi

echo "Found simulator UDID: $SIM_UDID"

SIM_DIR=~/Library/Logs/CoreSimulator/${SIM_UDID}/

if [ ! -d "$SIM_DIR" ]; then
    echo "Simulator directory $SIM_DIR does not exist!"
    exit 1
fi


echo "building app"
phonegap  build ios

echo "clearing simulator log"
SIM_LOG=${SIM_DIR}/system.log
echo "" > ${SIM_LOG}

echo "starting simulator with app"
ios-sim launch platforms/ios/build/emulator/MyPhoneGapTest.app --devicetypeid "${MOD_DEVICE}, ${SIM_IOS}" --exit

echo "waiting for tests"

tail -f ${SIM_LOG} | while read LOGLINE
do
    [[ "${LOGLINE}" =~ ^.*#.*spec.*assertion.*failure.* ]] && pkill -P $$ tail
done

tail ${SIM_LOG} | grep "^.*#.*specs.*assertions.*failures.*"
cp ${SIM_LOG} ios.log
`

In the first part of the script, I get the emulator's UDID via the given device type and iOS version. Then I use Phonegap only to build the app but start it then in ios-sim manually. In the last part, I only wait for the expected line.

I hope that this script may be useful for others, too! 

-- Max
